src/emoji/Emoji.js

```diff
@@ -1,6 +1,34 @@
-import { createIconSet } from 'react-native-vector-icons';
-import emojiMap from './emojiMap';
+import {createIconSet} from 'react-native-vector-icons';
+import emojiMap, { emoticonToEmojiMap, commonEmoticonList, textCodeList, emojiToTextCodeMap } from './emojiMap';
 
 const Icon = createIconSet(emojiMap, 'AppleColorEmoji');
 
 export default Icon;
+
+const getEmojiFromEmoticons = (emoticons: string): string => String.fromCodePoint(parseInt(emoticons, 16));
+
+export const replaceEmoticonToEmoji = (text: string) => {
```

Comment:
`replace` receives a regex and a replacer function. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

You should also tell the regex to search only at the end of the string.